### DISCUSSION TITLE
Is @requires_docker absolutely neccessary? Can we mock lambda without docker?

### DISCUSSION THREAD

**Discussion by rishacha** [spulec/moto#6985] [Q&A] [answered]
https://github.com/getmoto/moto/blob/bd71374c96329ba9c9b148069aaf808a2537924e/tests/test_sqs/test_sqs_integration.py#L13-L21

Hi, I was just wondering if `@requires_docker` as shown in the above test is absolutely required for mocking the lambda.
I am currently working on  client project using an AWS Workspace (running Windows) that doesn't have docker enabled.

I needed to mock a lambda trigger from sqs and I was just wondering if it's possible to test the trigger without docker - i.e. only using sqs and the `lambda_handler(event,ctx)` function.

**Comment by bblommers** [comment]
Short answer: No, this is not supported.

I do think it's a useful feature though, especially as a follow-up/extension to #5721, so I'll treat this as an enhancement request.

**Reply by bblommers** [reply]
I've created #6987 from this issue - please follow that for any updates.

**Comment by bblommers** [comment, marked as answer]
Hi @rishacha, this is now fixed in the upcoming v5 release.

All service-specific decorators are be replaced with a single `@mock_aws` decorator. To use the simple Lambda implementation everywhere, you can specify this in the configuration:
```
@mock_aws(config={"lambda": {"use_docker": False}})
def test():
    ...
```

See the full release announcement here: https://github.com/getmoto/moto/issues/7198

(Edited to clarify that V5 is now released).
